internal/service/workers/nooneisforgotten/main.go

```diff
@@ -131,23 +131,23 @@ func updateReferralUserEvents(db *pgdb.DB, types *evtypes.Types) error {
 	return nil
 }
 
-// claimReferralSpecificEvents claim fulfilled events for invited
-// friends which have passport scanned, if it possible
-func claimReferralSpecificEvents(db *pgdb.DB, types *evtypes.Types, levels config.Levels) error {
-	evType := types.Get(models.TypeReferralSpecific, evtypes.FilterInactive)
-	if evType == nil || !evType.AutoClaim {
+// autoClaimEvents claim fulfilled events which have auto-claim enabled. This is
+// useful if some events were inactive, then became active and must be claimed
+// automatically.
+func autoClaimEvents(db *pgdb.DB, types *evtypes.Types, levels config.Levels) error {
+	claimTypes := types.Names(evtypes.FilterByAutoClaim(true))
+	if len(claimTypes) == 0 {
 		return nil
 	}
 
 	events, err := pg.NewEvents(db).
-		FilterByType(models.TypeReferralSpecific).
+		FilterByType(claimTypes...).
 		FilterByStatus(data.EventFulfilled).
 		Select()
 	if err != nil {
-		return fmt.Errorf("failed to select passport scan events: %w", err)
+		return fmt.Errorf("failed to select fulfilled events: %w", err)
 	}
 
-	// we need to have maps which link nullifiers to events slice
 	nullifiers := make([]string, 0, len(events))
 	for _, event := range events {
 		nullifiers = append(nullifiers, event.Nullifier)
```

Comment:
I think you need to use map, because user can have two or more fulfilled event

Reply:
commented